src/server/protocol.d.ts

```diff
@@ -87,6 +87,28 @@ declare module ts.server.protocol {
         file: string;
     }
 
+    /** 
+      * Arguments for ProjectInfoResponse messages.
+      */
+    export interface ProjectInfoRequestArgs {
+        /**
+          * The file for the request (absolute pathname required).
+          */
+        file: string;
+        /**
+          * Indicate if the file name list of the project is needed
+          */
+        needFileNameList: boolean;

```

Comment:
When I try to build, I only need to know if the current project is a configured project or inferred one; and for configured project, I only need to know the path of the config file. Therefore the file list is not needed.